src/v/bytes/details/io_iterator_consumer.h

```diff
@@ -72,8 +72,7 @@ class io_iterator_consumer {
     template<typename Output>
     [[gnu::always_inline]] void consume_to(size_t n, Output out) {
         size_t c = consume(n, [&out](const char* src, size_t max) {
-            std::copy_n(src, max, out);
-            out += max;
+            out = std::ranges::copy_n(src, max, out).out;
```

Comment:
Doesn't 

```
out = std::copy_n(...)
```
also work? Not that using ranges is a problem but just wondering.

Reply:
ah right, I automatically went with ranges but yes also copy_n return the output iterator. I forgot about that. let me check if there are problems with std::copy_n, it's probably better to not use ranges here in case of backports